src/v/v8_engine/executor.cc

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "v8_engine/executor.h"
+
+#include "seastarx.h"
+#include "utils/mutex.h"
+
+#include <seastar/core/alien.hh>
+#include <seastar/core/do_with.hh>
+#include <seastar/core/future.hh>
+#include <seastar/core/gate.hh>
+#include <seastar/core/semaphore.hh>
+#include <seastar/core/sleep.hh>
+#include <seastar/core/smp.hh>
+#include <seastar/util/later.hh>
+
+#include <chrono>
+#include <memory>
+
+namespace v8_engine {
+
+namespace internal {
+
+// spsc_queue
+
+spsc_queue::spsc_queue()
+  : _items(ss::smp::count)
```

Comment:
why are we using this `ss::smp::count` as a max size of the queue ? Maybe make it configurable ?

Reply:
done